### ISSUE TITLE
code ERESOLVE Could not resolve dependency after Vaadin 23.0.16

### ISSUE THREAD

**Issue by sveine** [vaadin/flow#14662]
### Description of the bug

Cannot restart my project without a total clean.

On restart I get this error:

```
2022-09-27 14:07:50.849  INFO 17444 --- [onPool-worker-1] c.v.f.s.frontend.TaskUpdatePackages      : using 'C:\Program Files\nodejs\npm.cmd --no-update-notifier --no-audit --scripts-prepend-node-path=true --ignore-scripts install' for frontend package installation
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: mapbox-gl-leaflet@0.0.15
npm ERR! Found: leaflet@1.6.0
npm ERR! node_modules/leaflet
npm ERR!   leaflet@"file:frontend/core/local_node_modules/leaflet1.7.1-modified" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer leaflet@"file:frontend/core/local_node_modules/leaflet1.7.1-modified" from mapbox-gl-leaflet@0.0.15
npm ERR! node_modules/mapbox-gl-leaflet
npm ERR!   mapbox-gl-leaflet@"0.0.15" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: leaflet@undefined
npm ERR! node_modules/leaflet
npm ERR!   peer leaflet@"file:frontend/core/local_node_modules/leaflet1.7.1-modified" from mapbox-gl-leaflet@0.0.15
npm ERR!   node_modules/mapbox-gl-leaflet
npm ERR!     mapbox-gl-leaflet@"0.0.15" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See C:\Users\svein\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\svein\AppData\Local\npm-cache\_logs\2022-09-27T14_07_52_300Z-debug-0.log
```
![image](https://user-images.githubusercontent.com/367125/192550125-cd3db9e2-2d3d-4231-bcca-9d554dff8a54.png)

### Expected behavior

Work after restart as in 23.0.16

### Minimal reproducible example

Seems that the npm module "file:./frontend/core/local_node_modules/leaflet1.7.1-modified" cause the problem

### Versions

Hilla: 23.2.2
Flow: 23.2.2
Java: JetBrains s.r.o. 17.0.2
OS: amd64 Windows 10 10.0
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36
Live reload: Java active (HotswapAgent): Front end active

**Comment by sveine** [comment]
Minimal reproducible example (run twice)

[my-app.zip](https://github.com/vaadin/flow/files/9663114/my-app.zip)

**Comment by Artur-** [comment]
On a mac, running `mvn`, waiting for the app to start and then killing the app + running `mvn` again works as expected. No errors.

**Comment by sveine** [comment]
> On a mac, running `mvn`, waiting for the app to start and then killing the app + running `mvn` again works as expected. No errors.

Minimal reproducible example (run twice)
[my-app.zip](https://github.com/vaadin/flow/files/9663265/my-app.zip)

**Comment by Artur-** [comment]
Why are you running `npm install` as a postinstall command? That seems to be failing when you are running it multiple times. You should probably run it once and the info is then added to package.json

**Comment by sveine** [comment]
I use to clean up things by delete package.json, package-lock.json, node_modules... It was a issue to make postinstall work previously and this was working fine in 23.0.16. Is there any reason that this should stop working?

**Comment by Artur-** [comment]
I have no idea why you cannot install npm packages again and again in 23.2, maybe something with the npm version being upgraded

If you do not want to keep the package info in package.json, then you should be able to keep it as `@NpmPackage(value = "leaflet", version="./frontend/core/local_node_modules/leaflet1.7.1-modified")` instead

**Comment by sveine** [comment]
> If you do not want to keep the package info in package.json, then you should be able to keep it as `@NpmPackage(value = "leaflet", version="./frontend/core/local_node_modules/leaflet1.7.1-modified")` instead

Absolutely! This is the "perfect" solution. I have tried that i some previous versions of Vaadin, but it does not work:


java.lang.NumberFormatException: './frontend/core/local_node_modules/leaflet1.7.1-modified' is not a valid version!
	at com.vaadin.flow.server.frontend.FrontendVersion.<init>(FrontendVersion.java:160) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.FrontendVersion.<init>(FrontendVersion.java:128) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeUpdater.toVersion(NodeUpdater.java:584) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeUpdater.handleExistingVaadinDep(NodeUpdater.java:542) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeUpdater.addDependency(NodeUpdater.java:504) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.updatePackageJsonDependencies(TaskUpdatePackages.java:244) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.execute(TaskUpdatePackages.java:102) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:960) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:434) ~[vaadin-dev-server-23.2.2.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:349) ~[vaadin-dev-server-23.2.2.jar:na]

Tried with `file:` prefix also

java.lang.NumberFormatException: 'file:./frontend/core/local_node_modules/leaflet1.7.1-modified' is not a valid version!
	at com.vaadin.flow.server.frontend.FrontendVersion.<init>(FrontendVersion.java:160) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.FrontendVersion.<init>(FrontendVersion.java:128) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeUpdater.toVersion(NodeUpdater.java:584) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeUpdater.handleExistingVaadinDep(NodeUpdater.java:542) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeUpdater.addDependency(NodeUpdater.java:504) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.updatePackageJsonDependencies(TaskUpdatePackages.java:244) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.execute(TaskUpdatePackages.java:102) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:960) ~[flow-server-23.2.2.jar:23.2.2]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:434) ~[vaadin-dev-server-23.2.2.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:349) ~[vaadin-dev-server-23.2.2.jar:na]

**Comment by Artur-** [comment]
Created #14673 for that

**Comment by Artur-** [comment]
#14673 was fixed so not sure if there is anything else to do here

**Comment by sveine** [comment]
> #14673 was fixed so not sure if there is anything else to do here

I will try this in the next version of Vaadin and see if it works.

**Comment by mshabarov** [comment]
Fix for https://github.com/vaadin/flow/issues/14673 is available in V23.2.3.

**Comment by sveine** [comment]
This seems to be working now
